Networking/NetworkingTests/Mapper/CouponListMapperTests.swift

```diff
@@ -0,0 +1,122 @@
+        XCTAssertEqual(coupon.limitUsageToXItems, 10)
+        XCTAssertEqual(coupon.freeShipping, true)
+        XCTAssertEqual(coupon.productCategories, [123, 435, 232])
+        XCTAssertEqual(coupon.excludedProductCategories, [908])
+        XCTAssertEqual(coupon.excludeSaleItems, false)
+        XCTAssertEqual(coupon.minimumAmount, "5.00")
+        XCTAssertEqual(coupon.maximumAmount, "500.00")
+        XCTAssertEqual(coupon.emailRestrictions, ["*@a8c.com", "someone.else@example.com"])
+        XCTAssertEqual(coupon.usedBy, ["someone.else@example.com", "person@a8c.com"])
+    }
+
+    func test_CouponsList_map_accepts_nulls_in_expected_optional_fields() throws {
+        let coupons = try mapLoadMinimalCouponsResponse()
+        let coupon = coupons[0]
+
+        let dateFormatter = DateFormatter.Defaults.dateTimeFormatter
+
+        XCTAssertEqual(coupon.couponID, 10714)
+        XCTAssertEqual(coupon.code, "test")
+        XCTAssertEqual(coupon.amount, "0.00")
+        XCTAssertEqual(coupon.dateCreated, dateFormatter.date(from: "2021-04-13T08:26:25"))
+        XCTAssertEqual(coupon.dateModified, dateFormatter.date(from: "2021-04-13T08:26:25"))
+        XCTAssertEqual(coupon.discountType, .fixedCart)
+        XCTAssertEqual(coupon.description, "")
+        XCTAssertEqual(coupon.dateExpires, nil)
+        XCTAssertEqual(coupon.usageCount, 0)
+        XCTAssertEqual(coupon.individualUse, false)
+        XCTAssertEqual(coupon.productIds, [])
+        XCTAssertEqual(coupon.excludedProductIds, [])
+        XCTAssertEqual(coupon.usageLimit, nil)
+        XCTAssertEqual(coupon.usageLimitPerUser, nil)
+        XCTAssertEqual(coupon.limitUsageToXItems, nil)
+        XCTAssertEqual(coupon.freeShipping, false)
+        XCTAssertEqual(coupon.productCategories, [])
+        XCTAssertEqual(coupon.excludedProductCategories, [])
+        XCTAssertEqual(coupon.excludeSaleItems, false)
+        XCTAssertEqual(coupon.minimumAmount, "0.00")
+        XCTAssertEqual(coupon.maximumAmount, "0.00")
+        XCTAssertEqual(coupon.emailRestrictions, [])
+        XCTAssertEqual(coupon.usedBy, [])
+    }
+}
+
+
+// MARK: - Test Helpers
+///
+private extension CouponListMapperTests {
+
+    /// Returns the CouponListMapper output upon receiving `filename` (Data Encoded)
+    ///
+    func mapCoupons(from filename: String) throws -> [Coupon] {
+        guard let response = Loader.contentsOf(filename) else {
+            return []
+        }
+
+        return try CouponListMapper(siteID: dummySiteID).map(response: response)
+    }
+
+    /// Returns the CouponsMapper output from `coupons-all.json`
+    ///
+    func mapLoadAllCouponsResponse() throws -> [Coupon] {
+        return try mapCoupons(from: "coupons-all")
+    }
+
+    /// Returns the CouponsMapper output from `coupons-minimal.json`
+    ///
+    func mapLoadMinimalCouponsResponse() throws -> [Coupon] {
+        return try mapCoupons(from: "coupons-minimal")
+    }
```

Comment:
Is there a reason for having two JSON?

Reply:
This was so we could separate test the correct handling of nulls, but I could certainly add the JSON object from `coupons-minimal.json` to `coupons-all.json`. Only downside of doing this is that were the optional test to start failing, so would earlier tests unrelated to handling nulls correctly. That's not such a big deal, so if we'd prefer to minimise the number of JSON files I'll just move it over 😊

Reply:
I think that for this specific case you can move the coupon-minimal content as the third coupon in `coupons-all.json`